src/aiida/cmdline/commands/cmd_archive.py

```diff
@@ -521,6 +522,12 @@ def _import_archive_and_migrate(
                     )
             else:
                 _echo_exception(f'an exception occurred while trying to import the archive {archive}', exception)
+        except ImportTestRun:
+            echo.echo_success(
```

Comment:
I noticed that this will only cover the case for archives that are not migrated. That case will call `_import_archive` on line 518 and so you need to catch `ImportTestRun` as well on line 519. Here is a test that covers both cases:
```python
@pytest.mark.parametrize('archive', (
    get_archive_file('arithmetic.add.aiida', filepath='calcjob'),
    get_archive_file('export_0.9_simple.aiida', filepath=ARCHIVE_PATH),
))
def test_import_dry_run(run_cli_command, archive):
    """Test import dry-run"""
    result = run_cli_command(cmd_archive.import_archive, [archive, '--dry-run'])
    assert f'Dry-run of archive `{archive}` completed' in result.output
```

Reply:
Thanks, @sphuber, for the provided code snippet (which is basically the whole test suite needed for this :D). I removed the backticks surrounding the `{archive}`, as they are not used in the other f-strings that echo the archive, and made the sentence start lowercase (again, this seems to be the standard in `cmd_archive.py`; seems to be changing throughout the different `echo.echo_` statements during archive import).